metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -0,0 +1,125 @@
+package io.dockstore.metricsaggregator.helper;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import io.dockstore.metricsaggregator.Statistics;
+import io.dockstore.openapi.client.model.CpuMetric;
+import io.dockstore.openapi.client.model.Execution;
+import io.dockstore.openapi.client.model.ExecutionStatusMetric;
+import io.dockstore.openapi.client.model.ExecutionTimeMetric;
+import io.dockstore.openapi.client.model.MemoryMetric;
+import io.dockstore.openapi.client.model.Metrics;
+import io.dockstore.webservice.core.metrics.MemoryStatisticMetric;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AggregationHelper {
+    private static final Logger LOG = LoggerFactory.getLogger(AggregationHelper.class);
+
+    private AggregationHelper() {}
+
+    public static Optional<Metrics> getAggregatedMetrics(List<Execution> executions) {
+        Optional<ExecutionStatusMetric> aggregatedExecutionStatus = getAggregatedExecutionStatus(executions);
+        if (getAggregatedExecutionStatus(executions).isPresent()) {
+            Metrics aggregatedMetrics = new Metrics();
+            aggregatedMetrics.setExecutionStatusCount(aggregatedExecutionStatus.get());
+            getAggregatedExecutionTime(executions).ifPresent(aggregatedMetrics::setExecutionTime);
+            getAggregatedCpu(executions).ifPresent(aggregatedMetrics::setCpu);
+            getAggregatedMemory(executions).ifPresent(aggregatedMetrics::setMemory);
+            return Optional.of(aggregatedMetrics);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<ExecutionStatusMetric> getAggregatedExecutionStatus(List<Execution> executions) {
```

Comment:
It'd be helpful to me if each of these public methods had a short comment that explained what precisely was being calculated, and how.